### PR TITLE
[FIX] point_of_sale : Wrong price when using SO

### PR DESCRIPTION
**Pull request by robinengels**
Current behavior:
When changing unit price in the SO of a product tracked by SN
the price reset to the normal price when entering the SN in the PoS.

Expected behavior:
The modified price from the SO shouldn't be modified in the PoS when entering
the SN

Steps to reproduce:
Create a SO with a product tracked by SN and change the price of the product in the SO
Go to the PoS app and add this SO
Click on the product to add the SN
The price change

opw-2687001

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/js/models.js [hunk 1]
```diff
@@ -2205,7 +2205,11 @@ exports.Orderline = Backbone.Model.extend({
     get_unit_price: function(){
         var digits = this.pos.dp['Product Price'];
         // round and truncate to mimic _symbol_set behavior
-        return parseFloat(round_di(this.price || 0, digits).toFixed(digits));
+        let price = this.price
```
Comment (abz-odoo) [inline addons/point_of_sale/static/src/js/models.js:2208]:
```suggestion
        let price = this.price;
```

File: addons/point_of_sale/static/src/js/models.js [hunk 2]
```diff
@@ -2205,7 +2205,11 @@ exports.Orderline = Backbone.Model.extend({
...
+        if(this.sale_order_line_id){
```
Comment (abz-odoo) [inline addons/point_of_sale/static/src/js/models.js:2209]:
```suggestion
        if (this.sale_order_line_id){
```

File: addons/point_of_sale/static/src/js/models.js [hunk 3]
```diff
@@ -2205,7 +2205,11 @@ exports.Orderline = Backbone.Model.extend({
...
+            price = this.sale_order_line_id.price_total
```
Comment (abz-odoo) [inline addons/point_of_sale/static/src/js/models.js:2210]:
```suggestion
            price = this.sale_order_line_id.price_total;
```

**Review by abz-odoo — CHANGES_REQUESTED** [review]
Some codestyle suggestions 🙂 
see https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines

File: addons/point_of_sale/static/src/js/models.js [hunk 4]
```diff
@@ -2009,7 +2009,7 @@ exports.Orderline = Backbone.Model.extend({
         }
 
         // just like in sale.order changing the quantity will recompute the unit price
-        if(! keep_price && ! this.price_manually_set){
+        if(! keep_price && ! this.price_manually_set && ! this.sale_order_line_id){
```
Comment (hubvd) [inline addons/point_of_sale/static/src/js/models.js:2012]: I think it would be better to figure out why `this.price_manually_set` isn't true.

Comment (abz-odoo) [reply addons/point_of_sale/static/src/js/models.js:2012]:
```suggestion
        if (!keep_price && !this.price_manually_set && !this.sale_order_line_id){
```
same as above

**Comment by hubvd** [comment]
LGTM :+1:

**Review by nle-odoo — COMMENTED** [review]
with this scenario:

> Create a SO with a product tracked by SN and change the price of the product in the SO
> Go to the PoS app and add this SO
> __new__ refresh the page 
> Click on the product to add the SN
> The price change

the behavior is not fixed with the change (probably because we don't `export_as_JSON` and `import_as_JSON` the `price_manually_set` variable). If the behavior change is correct, I think this scenario should work the same as the one that is fixed.

as a side note, the line move make it seems like the variable will be used for import_as_JSON, but it isn't the case (so I would not move the line)

**Review by nle-odoo — APPROVED** [review]
ok to me if it's what we want

I think `this.price_manually_set = options.price_manually_set || false` should still be lower if it's not used when getting data from JSON but it's just a detail.

@pimodoo what do you think about this?

**Comment by pimodoo** [comment]
it looks good to me

**Comment by robinengels** [comment]
@pimodoo  @nle-odoo  can one of you R+ this ?

**Review by nle-odoo — COMMENTED** [review]
robodoo r+

**Pull request closed without merging**